p2p/tools/factories/peer.py

```diff
@@ -74,31 +75,8 @@
         do_handshake(bob),
     ), timeout=1)
 
-    alice_task = asyncio.ensure_future(alice.run())
-    bob_task = asyncio.ensure_future(bob.run())
-
-    await asyncio.wait_for(asyncio.gather(
-        alice.events.started.wait(),
-        bob.events.started.wait(),
-    ), timeout=1)
-
-    try:
+    async with run_service(alice), run_service(bob):
```

Comment:
Thoughts on making this a member method? Something like:
```suggestion
    async with alice.spawn(), bob.spawn():
```

Reply:
I'm gonna hold off for now on this.

https://github.com/ethereum/trinity/blob/b912a5ede37d6e9efc1137fc599561e4c7daed85/p2p/trio_service.py#L251

It keeps it congruent with the `trio` based service API which I like because it means less code changes when we switch from one to the other.